### DISCUSSION TITLE
Does DuckDB maintain the original order of records?

### DISCUSSION THREAD

**Discussion by laplab** [duckdb/duckdb#11507] [Q&A] [answered]
I have a Parquet file where order of records matters. I query this file using DuckDB through a query which looks like
```
SELECT field1, field2, ... FROM file.parquet WHERE <comparison predicates on fields>
```
No aggregations, no window functions, no `ORDER BY`. Am I guaranteed that DuckDB will return the records matching the predicate in the same order they appear in my Parquet file?

**Comment by Tishj** [comment, marked as answer]
<https://duckdb.org/docs/configuration/overview>
See the `preserve_insertion_order` configuration option

**Reply by soerenwolfers** [reply]
Does insertion order apply to parquet scans or Python datafames? It doesn't seem to apply to UNION operations
